src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrayOfStates = [];
+  const stateCopy = Object.assign({}, state);
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
You can destructure all the needed props of action (type, extraData, keysToRemove)